From 344773c4c6949757d9719850191229481c9733a9 Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Thu, 1 Sep 2016 17:25:23 +0200
Subject: [PATCH 40/79] SYSDB: Suppress warning from clang static analyser
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

scan-build wrongly assumes that output variable
"version" is not initialized if function sysdb_cache_connect
returns ERR_SYSDB_VERSION_TOO_OLD or ERR_SYSDB_VERSION_TOO_NEW

The reality is that output variable "version" is initialized
especially for these two case. Initialisation to NULL suppresses
these false positive reports.

Reviewed-by: Fabiano Fidêncio <fidencio@redhat.com>
(cherry picked from commit 3f6aecfe5061e165c10829142854ec7189029407)
---
 src/db/sysdb_init.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/db/sysdb_init.c b/src/db/sysdb_init.c
index d110aa7a2878e47650db177cfd342d0ac32248ab..538ba027cd94e274ba328d398cc565b11ea56f39 100644
--- a/src/db/sysdb_init.c
+++ b/src/db/sysdb_init.c
@@ -688,7 +688,7 @@ static int sysdb_domain_cache_connect(struct sysdb_ctx *sysdb,
                                       struct sysdb_dom_upgrade_ctx *upgrade_ctx)
 {
     errno_t ret;
-    const char *version;
+    const char *version = NULL;
     TALLOC_CTX *tmp_ctx;
     struct ldb_context *ldb;
 
-- 
2.9.3

